braindecode/models/lmda.py

```diff
@@ -0,0 +1,278 @@
+            ),
+            nn.BatchNorm2d(self.channel_depth_1),
+            nn.Conv2d(
+                in_channels=self.channel_depth_1,
+                out_channels=self.channel_depth_1,
+                kernel_size=(1, self.kernel_size_time),
+                groups=self.channel_depth_1,
+                bias=False,
+            ),
+            nn.BatchNorm2d(self.channel_depth_1),
+            self.activation(),
+        )
+
+        # Depth-wise Attention Module
+        self.depth_attention = _EEGDepthAttention(
+            num_channels=self.channel_depth_1,
+            num_times=reduced_time,
+            kernel_size=self.kernel_size_attention,
+        )
+
+        # Spatial Convolutional Layers
+        self.spatial_conv = nn.Sequential(
+            nn.Conv2d(
+                in_channels=self.channel_depth_1,
+                out_channels=self.channel_depth_2,
+                kernel_size=(1, 1),
+                groups=1,
+                bias=False,
+            ),
+            nn.BatchNorm2d(self.channel_depth_2),
+            nn.Conv2d(
+                in_channels=self.channel_depth_2,
+                out_channels=self.channel_depth_2,
+                kernel_size=(self.n_chans, 1),
+                groups=self.channel_depth_2,
+                bias=False,
+            ),
+            nn.BatchNorm2d(self.channel_depth_2),
+            self.activation(),
+        )
+
+        # Normalization Layers
+        self.normalization = nn.Sequential(
+            nn.AvgPool3d(kernel_size=(1, 1, self.avg_pool_size)),
+            nn.Dropout(p=self.drop_prob),
+        )
+
+        # Final Classification Layer
+        self.final_layer = nn.Linear(
+            in_features=embedding_size, out_features=self.n_outputs
+        )
+
+        # Initialize weights
+        self._initialize_weights()
+
+    def _initialize_weights(self):
+        """
+        Initialize weights of convolutional and linear layers.
+        """
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.xavier_uniform_(m.weight)
+                if m.bias is not None:
+                    nn.init.zeros_(m.bias)
+            elif isinstance(m, nn.BatchNorm2d):
+                nn.init.ones_(m.weight)
+                nn.init.zeros_(m.bias)
+            elif isinstance(m, nn.Linear):
+                nn.init.xavier_uniform_(m.weight)
+                if m.bias is not None:
+                    nn.init.zeros_(m.bias)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """
+        Forward pass of the LMDA model.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input tensor of shape `(batch_size, n_chans, time)`.
+
+        Returns
+        -------
+        torch.Tensor
+            Output logits of shape `(batch_size, n_outputs)`.
+
+        """
+        x = self.ensure_dim(x)
+        x = torch.einsum("bdcw, hdc->bhcw", x, self.channel_weight)  #
```

Comment:
```suggestion
        x = self.channel_weight(x)
```
With the convolution layer above, we would simplify to this expression.